### PR TITLE
Upgrade to v1.14.1

### PR DESCRIPTION
**Pull request by friedrichg**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,11 +3,11 @@ module github.com/cortexproject/cortex-tools
 go 1.22.2
 
 require (
-	cloud.google.com/go/storage v1.10.0
```
Comment (CharlieTLe) [inline go.mod:6]: This version has been retracted due to https://github.com/googleapis/google-cloud-go/issues/6857. Let's use a newer version that fixes the bug, e.g. `v1.40.0`

**Review by CharlieTLe — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -107,90 +116,103 @@ require (
...
-
 // Use fork of gocql that has gokit logs and Prometheus metrics.
 replace github.com/gocql/gocql => github.com/grafana/gocql v0.0.0-20200605141915-ba5dc39ece85
 
 // Using a 3rd-party branch for custom dialer - see https://github.com/bradfitz/gomemcache/pull/86
 replace github.com/bradfitz/gomemcache => github.com/themihai/gomemcache v0.0.0-20180902122335-24332e2d58ab
 
-// Required for Alertmanager
+// Replace memberlist with Grafana's fork which includes some fixes that haven't been merged upstream yet
+replace github.com/hashicorp/memberlist => github.com/grafana/memberlist v0.2.5-0.20211201083710-c7bc8e9df94b
```
Comment (CharlieTLe) [inline go.mod:203]: I don't think this is needed here.

File: go.mod [hunk 3]
```diff
@@ -107,90 +116,103 @@ require (
...
+
+// v0.20.5 caused go mod checksum error
+replace github.com/go-openapi/spec => github.com/go-openapi/spec v0.20.6
 
-replace github.com/hashicorp/consul => github.com/hashicorp/consul v1.8.1
+// the v6.0.5851 Prometheus depends on doesn't seem to exist anymore?
+replace github.com/ionos-cloud/sdk-go/v6 => github.com/ionos-cloud/sdk-go/v6 v6.0.4
 
-replace github.com/grafana-tools/sdk => github.com/colega/grafana-tools-sdk v0.0.0-20220323154849-711bca56d13f
+replace github.com/googleapis/gnostic => github.com/google/gnostic v0.6.9
```
Comment (CharlieTLe) [inline go.mod:218]: I don't think this is needed here.

File: go.mod [hunk 4]
```diff
@@ -107,90 +116,103 @@ require (
...
+
+// Same version being used by thanos
+replace github.com/vimeo/galaxycache => github.com/thanos-community/galaxycache v0.0.0-20211122094458-3a32041a1f1e
+
+// v0.20.5 caused go mod checksum error
+replace github.com/go-openapi/spec => github.com/go-openapi/spec v0.20.6
 
-replace github.com/hashicorp/consul => github.com/hashicorp/consul v1.8.1
+// the v6.0.5851 Prometheus depends on doesn't seem to exist anymore?
+replace github.com/ionos-cloud/sdk-go/v6 => github.com/ionos-cloud/sdk-go/v6 v6.0.4
```
Comment (CharlieTLe) [inline go.mod:216]: I don't think this is needed here.

File: go.mod [hunk 5]
```diff
@@ -107,90 +116,103 @@ require (
...
+
+// Replace grpc as weaveworks common needs to update this first
+// vendor/github.com/weaveworks/common/httpgrpc/server/server.go:137:3: undefined: grpc.WithBalancerName
+replace google.golang.org/grpc => google.golang.org/grpc v1.45.0
+
+// Same version being used by thanos
+replace github.com/vimeo/galaxycache => github.com/thanos-community/galaxycache v0.0.0-20211122094458-3a32041a1f1e
+
+// v0.20.5 caused go mod checksum error
+replace github.com/go-openapi/spec => github.com/go-openapi/spec v0.20.6
```
Comment (CharlieTLe) [inline go.mod:213]: I don't think this is needed here.

File: go.mod [hunk 6]
```diff
@@ -107,90 +116,103 @@ require (
...
 // Using a 3rd-party branch for custom dialer - see https://github.com/bradfitz/gomemcache/pull/86
 replace github.com/bradfitz/gomemcache => github.com/themihai/gomemcache v0.0.0-20180902122335-24332e2d58ab
 
-// Required for Alertmanager
+// Replace memberlist with Grafana's fork which includes some fixes that haven't been merged upstream yet
+replace github.com/hashicorp/memberlist => github.com/grafana/memberlist v0.2.5-0.20211201083710-c7bc8e9df94b
+
+// Replace grpc as weaveworks common needs to update this first
+// vendor/github.com/weaveworks/common/httpgrpc/server/server.go:137:3: undefined: grpc.WithBalancerName
+replace google.golang.org/grpc => google.golang.org/grpc v1.45.0
```
Comment (CharlieTLe) [inline go.mod:207]: I don't think this is needed here.

**Comment by friedrichg** [comment]
I am just trying to build according to https://github.com/cortexproject/cortex/blob/v1.14.1/go.mod#L216-L241
I will try your suggestions, let me double check them

File: go.mod (hunk 2 above)
Comment (friedrichg) [reply go.mod:203]: we are betting using the same version cortex uses 
https://github.com/cortexproject/cortex-tools/blob/f2909a3d92096cefbd5e0aaba26215af52e7f484/pkg/bench/ring_check.go#L14

**Comment by friedrichg** [comment]
yeah, no 
>vendor/github.com/weaveworks/common/httpgrpc/server/server.go:137:8: undefined: grpc.WithBalancerName
that is because of the grpc upgrade. I will revert

File: go.mod (hunk 1 above)
Comment (friedrichg) [reply go.mod:6]: it's the version used by cortex v1.14.1 https://github.com/cortexproject/cortex/blob/v1.14.1/go.mod#L78
will not touch this

**Pull request merged**